src/mlpack/methods/ann/layer/threshold_fn.hpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file methods/ann/layer/threshold_fn.hpp
+ * @author Shubham Agrawal
+ *
+ * Definition of the Threshold activation function
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_THRESHOLD_HPP
+#define MLPACK_METHODS_ANN_LAYER_THRESHOLD_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Threshold activation function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     x & : x > threshold \\
+ *     value & : x \le threshold
+ *   \end{array}
+ * \right.
+ * @f}
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class Threshold
+{
+ public:
```

Comment:
I do not know about this, I understand that the activation function is called `Threshold` but maybe we need to change the class name? I am not sure about this, @zoq  any thought?

Reply:
Can you suggest me another name?

Reply:
The name is fine for me.